src/main/java/com/example/userservice/service/UsersService.java

```diff
@@ -0,0 +1,123 @@
+        this.usersMapper=usersMapper;
+    }
+
+    public String body;
+
+    public User user;
+
+    public String phone;
+
+    public HttpStatus status;
+
+    public List<User> findByName(String nameStartsWith, String nameEndsWith){
+        if (nameStartsWith != null && nameEndsWith != null) {
+            return usersMapper.findByNameStartingAndEnding(nameStartsWith, nameEndsWith);
+        }
+        else if (nameStartsWith != null) {
+            return usersMapper.findByNameStartingWith(nameStartsWith);
+        }
+        else if (nameEndsWith != null) {
+            return usersMapper.findByNameEndingWith(nameEndsWith);
+        }
+        else {
+            return usersMapper.findAll();
+        }
+    }
+
+    public User findUser(int id){
+        return usersMapper.findById(id).orElseThrow(() -> new UserNotFoundException("User not found"));
+    }
+
+    public User userInsert(String name,String phone) {
+        Optional<User> userOptional=usersMapper.findByPhone(phone);
+
+        if ((name == null || name.isEmpty()) && (phone == null  || phone.isEmpty())) {
+            throw new  BadRequestException("name and phone cannot be null");
+        }
+        else if (name == null || name.isEmpty()) {
+            throw new  BadRequestException("name cannot be null");
+        }
+        else if (phone == null || phone.isEmpty()) {
+            throw new  BadRequestException("phone cannot be null");
+        }
+        else if (userOptional.isPresent()) {
+            throw new  BadRequestException("The phone is being used");
+        }
+        else {
+            this.user = new User(name, phone);
+            usersMapper.insert(user);
+            this.body="user created";
+            return user;
+        }
+    }
+
+    public User userPatch(int id,String name,String phone) {
+        Optional<User> updateUser = usersMapper.findById(id);
+
+        if (updateUser.isPresent()) {
+            if ((name == null || phone == null ) || (name.isEmpty() && phone.isEmpty())) {
+                throw new  BadRequestException("name and phone cannot be null");
+            }
+            else if (name.isEmpty()) {
+                this.user = new User(id,name, phone);
+                usersMapper.updatePhone(user);
+                this.body="user's phone updated";
+                return user;
+            }
+            else if (phone.isEmpty()) {
+                this.user = new User(id,name, phone);
+                usersMapper.updateName(user);
+                this.body="user's name updated";
+                return user;
+            }
+            else{
+                this.user = new User(id,name, phone);
+                usersMapper.update(user);
+                this.body="user updated";
+                return user;
+            }
+        }
+        else {
+            throw new UserNotFoundException("User not found");
+        }
+    }
+
+    public User userDelete(int id){
```

Comment:
こちらの方が英語としてもしっくり来ますね
```suggestion
    public User deleteUser(int id){
```

Reply:
https://github.com/Ouichi/Task10/blob/0bb12a761f4ccb3cc4e5a621f676ee54655a5154/src/main/java/com/example/userservice/service/UsersService.java#L99